File: frontend/src/api/types/spark.ts
export type SparkTrigger = "wrong_answer" | "low_confidence" | "hint_request" | "free_chat";

export interface SparkConversationCreate {
  student_id: string;
  question_id?: string;
  trigger: SparkTrigger;
  competency_id?: string;
  selected_option?: string;
  confidence_report?: string;
}

export interface SparkTurnRequest {
  content: string;
}

export interface SparkHintRequest {
  student_id: string;
  question_id: string;
}

export interface SparkMessageOut {
  role: "spark" | "student" | "system";
  content: string;
  created_at: string;
}

export interface SparkStartResponse {
  conversation_id: string;
  message: SparkMessageOut;
}

export interface SparkTurnResponse {
  message: SparkMessageOut;
  evidence_submitted: boolean;
  is_complete: boolean;
}

export interface SparkEndResponse {
  message: SparkMessageOut;
  evidence_submitted: boolean;
}

export interface SparkHintResponse {
  hint: string;
}

File: backend/app/schemas/spark.py
"""SPARK API request/response schemas."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


# ── Requests ──


class SparkConversationCreate(BaseModel):
    student_id: UUID
    question_id: UUID | None = None
    trigger: str = Field(pattern=r"^(wrong_answer|low_confidence|hint_request|free_chat)$")
    competency_id: str | None = None
    selected_option: str | None = None
    confidence_report: str | None = None
    student_response: dict | None = None  # raw student answer for contextual hints


class SparkTurnRequest(BaseModel):
    content: str = Field(min_length=1, max_length=2000)


class SparkHintRequest(BaseModel):
    student_id: UUID
    question_id: UUID
    student_response: dict | None = None


# ── Responses ──


class SparkMessageOut(BaseModel):
    role: str
    content: str
    created_at: datetime

    model_config = {"from_attributes": True}


class SparkConversationOut(BaseModel):
    id: UUID
    student_id: UUID
    trigger: str
    status: str
    competency_id: str | None
    evidence_submitted: bool
    created_at: datetime
    ended_at: datetime | None

    model_config = {"from_attributes": True}


class SparkStartResponse(BaseModel):
    conversation_id: UUID
    message: SparkMessageOut


class SparkTurnResponse(BaseModel):
    message: SparkMessageOut
    evidence_submitted: bool
    is_complete: bool


class SparkEndResponse(BaseModel):
    message: SparkMessageOut
    evidence_submitted: bool


class SparkHintResponse(BaseModel):
    hint: str


class SparkConversationDetailOut(BaseModel):
    conversation: SparkConversationOut
    messages: list[SparkMessageOut]
